src/sdk/object_sdk.js

```diff
@@ -233,9 +233,10 @@ class ObjectSDK {
             ns_info.endpoint_type === 'S3_COMPATIBLE' ||
             ns_info.endpoint_type === 'FLASHBLADE' ||
             ns_info.endpoint_type === 'IBM_COS') {
-            const httpOptions = (ns_info.endpoint_type === 'AWS') ? undefined : {
-                agent: http_utils.get_unsecured_http_agent(ns_info.endpoint)
-            };
+
+            const agent = http_utils.get_http_agent(ns_info.endpoint, {
+                rejectUnauthorized: ns_info.endpoint_type !== 'AWS'
```

Comment:
you have a similar check inside get_http_agent to detect aws endpoints.
why do you have it in both?

Reply:
It is not exactly the same. 
This condition checks the `endpoint_type` while the other checks the actual endpoint address. I didn't want to take the chance and covered the case where we have an AWS type is set but not an endpoint with `amazon.com` in it. 
You might call it defensive coding

Also, I took this approach because this check was there even before the change which made me assume that the above case might happen.

Reply:
I would leave this in one place - in object_sdk in my opinion.
because finding such a condition in a generic http_utils is quite a surprising side effect.

Reply:
keep the util passive, and the caller can choose secure/unsecure according to specific needs.

Reply:
I see your point, and I was also surprised to see that inside the original utils function, but removing this from the utils code might have side effects when calling sites assumes that amazon will always be secured.
Analyzing every calling site for the possibility that an `amazon.com` address might be used there should be its own task that we should target for a later PR. This PR is mainly for resolving the issue attached in the description.

Reply:
@guymguym can this be resolved?